pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1688,8 +1697,11 @@ base type (-1 if base type does not use a typmod). -1 if this type is not a doma
       }
       tuple[22] = connection.encodeString(autoinc); // IS_AUTOINCREMENT
 
-      // there is no way to tell if the value was actually autogenerated.
-      tuple[23] = connection.encodeString(""); // IS_AUTOGENERATED
+      String generatedcolumn = "NO";
+      if (generated != null) {
```

Comment:
what happens if generated is not null but  false here ?

Reply:
The **generated** variable is of type String, so it can't be false.

Based on the information on this page:

https://www.postgresql.org/docs/13/catalog-pg-attribute.html

This is the documentation for the attgenerated column:

attgenerated char

If a zero byte (''), then not a generated column. Otherwise, s = stored. (Other values might be added in the future.)

The logic I added when retrieving the attgenerated column:

```
    if (connection.haveMinimumServerVersion(ServerVersion.v12)) {
      sql += "nullif(a.attgenerated, '') as attgenerated,";
    } else {
      sql += "null as attgenerated,";
    }
```
Will convert the value to null when it is an empty string.

The logic I added is similar to what was done to determine if a column is an identity column.

Reply:
sounds good